cpp/email_watcher.cc

```diff
@@ -0,0 +1,258 @@
+/** \file    email_watcher.cc
+ *  \brief   Checks an mbox file for overdue and urgent messages.
+ *  \author  Dr. Johannes Ruscheinski
+ */
+
+/*
+    Copyright (C) 2020 Library of the University of Tübingen
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Affero General Public License as
+    published by the Free Software Foundation, either version 3 of the
+    License, or (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Affero General Public License for more details.
+
+    You should have received a copy of the GNU Affero General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <map>
+#include <memory>
+#include <set>
+#include <unordered_map>
+#include <cstdlib>
+#include "DbConnection.h"
+#include "EmailSender.h"
+#include "FileUtil.h"
+#include "IniFile.h"
+#include "MBox.h"
+#include "RegexMatcher.h"
+#include "StringUtil.h"
+#include "TimeUtil.h"
+#include "UBTools.h"
+#include "util.h"
+
+
+namespace {
+
+
+struct EmailDescription {
+    std::string from_host_;
+    RegexMatcher *subject_pattern_;
```

Comment:
subject_pattern_ and email_body_pattern_ do not contain a pattern => they contain a matcher, so the variables should be named subject_matcher_ and email_body_matcher_

Reply:
and maybe they should be unified to subject_matcher_ + body_matcher_ or email_subject_matcher_ + email_body_matcher_